src/Network/Wai/Middleware/OpenApi/PathMap.hs

```diff
@@ -17,17 +19,21 @@ import Data.OpenApi (OpenApi, PathItem)
 import Data.OpenApi qualified as OpenApi
 import System.FilePath.Posix qualified as Posix
 
-newtype PathMap = PathMap
-  { unwrap :: Map TemplatedPath PathItem
+newtype PathMap a = PathMap
```

Comment:
Is this polymorphism just to make the tests easier to write?

FWIW, I find the property-based testing hard to follow and not very good at documenting the behaviors you uncovered. I think things would be clearer if you gave yourself a helper to create concrete `PathItem`s and test against fixed scenarios with them.

Reply:
> Is this polymorphism just to make the tests easier to write?

Yeah, it means you don't actually have to construct `PathItem`s, since that's not relevant to what's being tested, which is only how the keys match. But we could give the polymorphic type a different name and alias `type PathMap = Something PathItem` to avoid complicating the user API.

> I find the property-based testing hard to follow

Yeah, I just started down that road because I was suspecting that I needed quickcheck to generate some larger maps for me to find the problem cases. If wonky `Ord`s are a problem I don't necessarily expect to be able to find it with small examples.

Reply:
> means you don't actually have to construct `PathItem`s

Well, you have to define something you can distinguish. I don't really see `mempty & summary ?~ "foo"` _that_ much more difficult than some other non-`PathItem` type (that can be distinguished). And for test that needn't distinguish (like an `isNothing` assertion), then `mempty` is about as simple as it gets.